### PR TITLE
Support setting customized envs for backup and restore jobs

### LINKED ISSUE #3831: Support setting customized envs for backup and restore jobs
## Feature Request

**Is your feature request related to a problem? Please describe:**

We have met some issues about backuping via br. Some of them are related to go runtime and can be mitigated by setting some GO env vars.
For example:
GODEBUG=madvdontneed=1 

**Describe the feature you'd like:**

Support setting env vars for backup and restore jobs.

**Describe alternatives you've considered:**

N/A

**Teachability, Documentation, Adoption, Migration Strategy:**

It's more like an internal debugging API, it should not be documented/release publicly.

### PR DESCRIPTION
**Pull request by dragonly**
### What problem does this PR solve?

Closes #3831

### What is changed and how does it work?

Add `Env` field in `BackupSpec`, and add logic/ut cases for backup, restore and clean jobs.

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Support setting customized envs for backup and restore jobs
```

### CODE REVIEW

File: pkg/backup/backup/backup_manager.go [hunk 1]
```diff
@@ -355,6 +355,20 @@ func (bm *backupManager) makeBackupJob(backup *v1alpha1.Backup) (*batchv1.Job, s
 		Value: string(rune(1)),
 	})
 
+	// set env vars specified in backup.Spec.Env if not present
+	for _, backupEnv := range backup.Spec.Env {
```
Comment (DanielZhangQD) [inline pkg/backup/backup/backup_manager.go:359]: What about using the function here https://github.com/pingcap/tidb-operator/blob/master/pkg/util/util.go#L256?

**Comment by DanielZhangQD** [comment]
@dragonly could you please add the test result?

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 2]
```diff
@@ -1282,10 +1282,13 @@ const (
 	CleanPolicyTypeDelete CleanPolicyType = "Delete"
 )
 
-// +k8s:openapi-gen=true
 // BackupSpec contains the backup specification for a tidb cluster.
+// +k8s:openapi-gen=true
 type BackupSpec struct {
 	corev1.ResourceRequirements `json:"resources,omitempty"`
+	// List of environment variables to set in the container, like v1.Container.Env.
```
[the remark this answers is not in the document]
Comment (dragonly) [reply pkg/apis/pingcap/v1alpha1/types.go:1289]: done

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by handlerww — APPROVED** [review]
LGTM

**Pull request merged**